qa: Intermittent failure in `feature_segwit.py --legacy-wallet`

On master (a8bbd4cc819633ec50ed0f763b6a75330ae055fb), native Windows [build](https://api.cirrus-ci.com/v1/task/4854921808838656/logs/functional_tests.log):
```
8/219 - feature_segwit.py --legacy-wallet failed, Duration: 17 s
stdout:
2021-09-28T08:10:35.808000Z TestFramework (INFO): Initializing test directory C:\Users\ContainerAdministrator\AppData\Local\Temp\test_runner_₿_🏃_20210928_080804\feature_segwit_208
2021-09-28T08:10:39.302000Z TestFramework (INFO): Verify sigops are counted in GBT with pre-BIP141 rules before the fork
2021-09-28T08:10:49.277000Z TestFramework (INFO): Verify witness txs are skipped for mining before the fork
2021-09-28T08:10:50.441000Z TestFramework (INFO): Verify unsigned p2sh witness txs without a redeem script are invalid
2021-09-28T08:10:50.488000Z TestFramework (INFO): Verify previous witness txs skipped for mining can now be mined
2021-09-28T08:10:50.535000Z TestFramework (INFO): Verify default node can't accept txs with missing witness
2021-09-28T08:10:50.581000Z TestFramework (INFO): Verify block and transaction serialization rpcs return differing serializations depending on rpc serialization flag
2021-09-28T08:10:50.738000Z TestFramework (INFO): Verify witness txs without witness data are invalid after the fork
2021-09-28T08:10:50.769000Z TestFramework (INFO): Verify default node can now use witness txs
2021-09-28T08:10:50.960000Z TestFramework (INFO): Verify sigops are counted in GBT with BIP141 rules after the fork
2021-09-28T08:10:51.971000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 131, in main
    self.run_test()
  File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\feature_segwit.py", line 239, in run_test
    assert tmpl['transactions'][0]['sigops'] == 8
AssertionError
```

With `assert_equal` https://api.cirrus-ci.com/v1/task/4880559223603200/logs/functional_tests.log:
```
9/219 - feature_segwit.py --legacy-wallet failed, Duration: 18 s
stdout:
2021-09-28T12:44:47.048000Z TestFramework (INFO): Initializing test directory C:\Users\ContainerAdministrator\AppData\Local\Temp\test_runner_₿_🏃_20210928_124215\feature_segwit_208
2021-09-28T12:44:51.331000Z TestFramework (INFO): Verify sigops are counted in GBT with pre-BIP141 rules before the fork
2021-09-28T12:45:00.525000Z TestFramework (INFO): Verify witness txs are skipped for mining before the fork
2021-09-28T12:45:01.751000Z TestFramework (INFO): Verify unsigned p2sh witness txs without a redeem script are invalid
2021-09-28T12:45:01.797000Z TestFramework (INFO): Verify previous witness txs skipped for mining can now be mined
2021-09-28T12:45:02.861000Z TestFramework (INFO): Verify default node can't accept txs with missing witness
2021-09-28T12:45:02.908000Z TestFramework (INFO): Verify block and transaction serialization rpcs return differing serializations depending on rpc serialization flag
2021-09-28T12:45:03.080000Z TestFramework (INFO): Verify witness txs without witness data are invalid after the fork
2021-09-28T12:45:03.126000Z TestFramework (INFO): Verify default node can now use witness txs
2021-09-28T12:45:03.298000Z TestFramework (INFO): Verify sigops are counted in GBT with BIP141 rules after the fork
2021-09-28T12:45:04.126000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 131, in main
    self.run_test()
  File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\feature_segwit.py", line 244, in run_test
    assert_equal(tmpl['transactions'][0]['sigops'], 8)
  File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\util.py", line 49, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(9 == 8)
```

Managed to get some more output locally using a few extra log commands:

```
2021-09-29T04:12:35.828000Z TestFramework (INFO): [RAWTX]:
2021-09-29T04:12:35.829000Z TestFramework (INFO): {'txid': '437738c723bef814efaa4ea6c1d75abad7c5a46581193ee45752206458bdd5ae', 'hash': '42fdc6f5599720870fc0dc4c7afcaa39bbd568b1b1c6485cacc5880443b58eff', 'version': 2, 'size': 268, 'vsize': 183, 'weight': 730, 'locktime': 436, 'vin': [{'txid': 'd383c5ede4a367911c11614c51f462f78a66efe16755410dc0e318e534ddf3a4', 'vout': 0, 'scriptSig': {'asm': '0020366c4902213bcdb36c442b74b4de819ab4afb7ffdf40764d4e7f9eb9aa1c6e4f', 'hex': '220020366c4902213bcdb36c442b74b4de819ab4afb7ffdf40764d4e7f9eb9aa1c6e4f'}, 'txinwitness': ['', '30440220085d600265bb5bd8b5dcb37dccf0bdcb5a17cb557f7915c5a6bbfd6fc92f25ee022054938052d9252314f81f5c5e87d3c6a71f290a0e0f605a2c3b8fbb664ebb295401', '5121039e8d8e9d1b2510cef293407a8de9183cabcb15ec694344b9a826a56c5dde400551ae'], 'sequence': 4294967294}], 'vout': [{'value': Decimal('48.99896340'), 'n': 0, 'scriptPubKey': {'asm': 'OP_DUP OP_HASH160 31695740fef5116954ddad139a9c9e1c6dc530e4 OP_EQUALVERIFY OP_CHECKSIG', 'hex': '76a91431695740fef5116954ddad139a9c9e1c6dc530e488ac', 'address': 'mk2DbfRWzFEuyXnpuDvHKyjGTTcCUitPJZ', 'type': 'pubkeyhash'}}, {'value': Decimal('1.00000000'), 'n': 1, 'scriptPubKey': {'asm': 'OP_DUP OP_HASH160 96538a8277e4ef9403ad33bb4191f88e95c08475 OP_EQUALVERIFY OP_CHECKSIG', 'hex': '76a91496538a8277e4ef9403ad33bb4191f88e95c0847588ac', 'address': 'muDojMzU1fg6DEVWFPRJxbU4g8LFgVVz1o', 'type': 'pubkeyhash'}}], 'hex': '0200000001a4f3dd34e518e3c00d415567e1ef668af762f4514c61111c9167a3e4edc583d30000000023220020366c4902213bcdb36c442b74b4de819ab4afb7ffdf40764d4e7f9eb9aa1c6e4ffeffffff02147c0e24010000001976a91431695740fef5116954ddad139a9c9e1c6dc530e488ac00e1f505000000001976a91496538a8277e4ef9403ad33bb4191f88e95c0847588acb4010000'}
2021-09-29T04:12:35.831000Z TestFramework (INFO): [TEMPLATE]:
2021-09-29T04:12:35.831000Z TestFramework (INFO): {'capabilities': ['proposal'], 'version': 536870912, 'rules': ['csv', '!segwit', 'testdummy', 'taproot'], 'vbavailable': {}, 'vbrequired': 0, 'previousblockhash': '1f50adae7f37543549ffe6e22f236eceb33355c06f23699d6c6bc0002b590e83', 'transactions': [{'data': '02000000000101a4f3dd34e518e3c00d415567e1ef668af762f4514c61111c9167a3e4edc583d30000000023220020366c4902213bcdb36c442b74b4de819ab4afb7ffdf40764d4e7f9eb9aa1c6e4ffeffffff02147c0e24010000001976a91431695740fef5116954ddad139a9c9e1c6dc530e488ac00e1f505000000001976a91496538a8277e4ef9403ad33bb4191f88e95c0847588ac03004730440220085d600265bb5bd8b5dcb37dccf0bdcb5a17cb557f7915c5a6bbfd6fc92f25ee022054938052d9252314f81f5c5e87d3c6a71f290a0e0f605a2c3b8fbb664ebb295401255121039e8d8e9d1b2510cef293407a8de9183cabcb15ec694344b9a826a56c5dde400551aeb4010000', 'txid': '437738c723bef814efaa4ea6c1d75abad7c5a46581193ee45752206458bdd5ae', 'hash': '42fdc6f5599720870fc0dc4c7afcaa39bbd568b1b1c6485cacc5880443b58eff', 'depends': [], 'fee': 3660, 'sigops': 9, 'weight': 730}], 'coinbaseaux': {}, 'coinbasevalue': 1250003660, 'longpollid': '1f50adae7f37543549ffe6e22f236eceb33355c06f23699d6c6bc0002b590e83570', 'target': '7fffff0000000000000000000000000000000000000000000000000000000000', 'mintime': 1632888824, 'mutable': ['time', 'transactions', 'prevblock'], 'noncerange': '00000000ffffffff', 'sigoplimit': 80000, 'sizelimit': 4000000, 'weightlimit': 4000000, 'curtime': 1632888824, 'bits': '207fffff', 'height': 437, 'default_witness_commitment': '6a24aa21a9ed2d0140b70952200db4620b5477ef45815e135c1f432f8323d34ee989ea47393b'}
2021-09-29T04:12:35.831000Z TestFramework (INFO): [TX]:
2021-09-29T04:12:35.831000Z TestFramework (INFO): 437738c723bef814efaa4ea6c1d75abad7c5a46581193ee45752206458bdd5ae
2021-09-29T04:12:35.831000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/Users/sdob814/git/bitcoin/test/functional/test_framework/test_framework.py", line 131, in main
    self.run_test()
  File "./test/functional/feature_segwit.py", line 251, in run_test
    assert_equal(tmpl['transactions'][0]['sigops'], 8)
  File "/Users/sdob814/git/bitcoin/test/functional/test_framework/util.py", line 49, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(9 == 8)

```

--- a/test/functional/feature_segwit.py
+++ b/test/functional/feature_segwit.py
@@ -236,12 +236,14 @@ class SegWitTest(BitcoinTestFramework):
 
         self.log.info("Verify sigops are counted in GBT with BIP141 rules after the fork")
         txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 1)
+        raw_tx = self.nodes[0].getrawtransaction(txid, True)
         tmpl = self.nodes[0].getblocktemplate({'rules': ['segwit']})
         assert_greater_than_or_equal(tmpl['sizelimit'], 3999577)  # actual maximum size is lower due to minimum mandatory non-witness data
         assert_equal(tmpl['weightlimit'], 4000000)
         assert_equal(tmpl['sigoplimit'], 80000)
         assert_equal(tmpl['transactions'][0]['txid'], txid)
-        assert_equal(tmpl['transactions'][0]['sigops'], 8)
+        expected_sigops = 9 if 'txinwitness' in raw_tx["vin"][0] else 8
+        assert_equal(tmpl['transactions'][0]['sigops'], expected_sigops)
         assert '!segwit' in tmpl['rules']
 
         self.generate(self.nodes[0], 1)  # Mine a block to clear the gbt cache

